File: Modules/Sessions/Models/SessionAttendance.php
<?php

namespace Modules\Sessions\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Modules\User\Models\User;

class SessionAttendance extends Model
{
    protected $fillable = [
        'session_id',
        'user_id',
        'joined_at',
        'left_at',
        'was_present',
        'rating',
        'comment',
    ];

    protected $casts = [
        'joined_at' => 'datetime',
        'left_at' => 'datetime',
        'was_present' => 'boolean',
        'rating' => 'integer',
    ];

    public function session(): BelongsTo
    {
        return $this->belongsTo(Session::class);
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}

File: Solid.Api/Database/Entities/SessionAttendance.cs
namespace Solid.Api.Database.Entities;

public sealed class SessionAttendance
{
    public long Id { get; set; }

    public long SessionId { get; set; }

    public long UserId { get; set; }

    public DateTime? JoinedAt { get; set; }

    public DateTime? LeftAt { get; set; }

    public bool WasPresent { get; set; }

    public byte? Rating { get; set; }

    public string? Comment { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public TherapySession Session { get; set; } = null!;

    public User User { get; set; } = null!;
}
